frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/FeatureOverviewEnvironment.tsx

```diff
@@ -135,115 +75,70 @@ const FeatureOverviewEnvironment = ({
-                                environmentMetric={environmentMetric}
-                                disabled={!env.enabled}
-                            />
-                        </StyledAccordionSummary>
-
-                        <StyledAccordionDetails enabled={env.enabled}>
-                            <StyledEnvironmentAccordionBody
-                                featureEnvironment={featureEnvironment}
-                                isDisabled={!env.enabled}
-                                otherEnvironments={feature?.environments
-                                    .map(({ name }) => name)
-                                    .filter((name) => name !== env.name)}
-                            />
-                            <ConditionallyRender
-                                condition={
-                                    (featureEnvironment?.strategies?.length ||
-                                        0) > 0
-                                }
-                                show={
-                                    <>
-                                        <Box
-                                            sx={{
-                                                display: 'flex',
-                                                justifyContent: 'center',
-                                                py: 1,
-                                            }}
-                                        >
-                                            <FeatureStrategyMenu
-                                                label='Add strategy'
-                                                projectId={projectId}
-                                                featureId={featureId}
-                                                environmentId={env.name}
-                                            />
-                                        </Box>
-                                        <EnvironmentFooter
-                                            environmentMetric={
-                                                environmentMetric
-                                            }
-                                        />
-                                        {showChangeRequestUpgrade ? (
-                                            <UpgradeChangeRequests />
-                                        ) : null}
-                                    </>
-                                }
-                            />
-                        </StyledAccordionDetails>
-                    </StyledAccordion>
-                </StyledFeatureOverviewEnvironment>
-            }
-        />
+        <StyledFeatureOverviewEnvironment>
+            <StyledAccordion
+                TransitionProps={{ mountOnEnter: true }}
+                data-testid={`${FEATURE_ENVIRONMENT_ACCORDION}_${env.name}`}
+                expanded={isOpen && hasActivations}
+                onChange={() => setIsOopen(isOpen ? !isOpen : hasActivations)}
```

Comment:
We should also double check the rendered component here. Again, according to the [MUI docs](https://mui.com/material-ui/react-accordion/#changing-heading-level):

> By default, the Accordion uses an h3 element for the heading. You can change the heading element using the slotProps.heading.component prop to ensure the correct heading hierarchy in your document.

I think h2 is right here (flag -> environment):

```tsx
<Accordion slotProps={{ heading: { component: 'h2' } }}>
```

But we should also make sure that there's no invalid components inside the h2, so checking the rendered hierarchy is probably smart.

Reply:
That's a link to MUI v6. Right now we have `div`s all the way down

Reply:
You're right! In that case, I'm assuming you've assessed the situation and checked the header levels, then. Resolving.